components/board/genericlinux/pin_types.go

```diff
@@ -26,11 +27,31 @@ type PinDefinition struct {
 	PWMID               int    // -1 for none
 }
 
+// board types have to be below so non-linux devices will compile
```

Comment:
I marked data.go with the linux flag so these structs need to be moved to a general file otherwise it raises an error in the mac build is what i meant, i can remove the comment